client_test.go

```diff
@@ -95,6 +95,18 @@ func TestDoDecodesTheJsonResponseIntoAStruct(t *testing.T) {
 	assert.Equal(t, "Hello World", data.Response.Message, "JSON was parsed correctly")
 }
 
+func TestSetBaseURL(t *testing.T) {
+	client := NewClient("project-id", "api-key", "test-application/0.0.1")
+
+	url := "https://dev-api.glesys.local"
```

Comment:
```suggestion
	url := "https://dev-api.glesys.test"
```

The `.local` TLD is reserved and may not be routed but it is still a valid TLD for instance in link local networking. In contrast the `.test` TLD is reserved for use when testing sorftware see [RFC 2606.](https://www.rfc-editor.org/rfc/rfc2606.html#page-2).